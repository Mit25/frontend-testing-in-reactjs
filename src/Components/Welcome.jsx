import $ from 'jquery';
import React from 'react';
import ExternalClient from '../Commons/ExternalClient';

export default class Welcome extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            uname: "Guest"
        }
    }

    componentDidMount() {
        var self = this;
        $.when(ExternalClient.getUsername())
            .done(function(uname) {
                self.setState({
                    uname: uname
                })
            }).fail(function() {
                self.setState({
                    uname: "Stranger"
                })
            });
    }

    render() {
        return (
            <h2>Hi, {this.state.uname}!!!</h2>
        )
    }
}

Welcome.defaultProps = {

};