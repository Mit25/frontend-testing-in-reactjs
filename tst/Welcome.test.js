import $ from 'jquery';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Welcome from '../src/Components/Welcome';

describe('Counter component', () => {

    it('snapshot testing for Welcome component', () => {
        const tree = renderer.create(<Welcome />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render Welcome component with shallow', () => {
        const wrapper = shallow(<Welcome />);

        expect(wrapper.find('h2').text()).toBe('Hi, Guest!!!');
    });

    it('should render Welcome component getUsername success', () => {
        const wrapper = shallow(<Welcome />);
        const uname = "MyName";
        $.when.mockImplementation(() => {
            return $.Deferred().resolve(uname);
        });
        wrapper.instance().forceUpdate();

        expect(wrapper.find('h2').text()).toBe('Hi, ' + uname + '!!!');
    });

    it('should render Welcome component with shallow', () => {
        const wrapper = shallow(<Welcome />);
        $.when.mockImplementation(() => {
            return $.Deferred().reject();
        });
        wrapper.instance().forceUpdate();

        expect(wrapper.find('h2').text()).toBe('Hi, Stranger!!!');
    });
});