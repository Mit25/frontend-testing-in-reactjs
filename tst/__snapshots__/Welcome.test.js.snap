// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component snapshot testing for Welcome component 1`] = `
<h2>
  Hi, 
  Guest
  !!!
</h2>
`;
